lms/djangoapps/sga_api/views.py

```diff
@@ -0,0 +1,276 @@
+            return Response(data={"error_message": "User already exists"}, status=status.HTTP_409_CONFLICT)
+
+        try:
+            data['password'] = uuid4().hex
+            data['country'] = country
+            user = create_account_with_params(request, data)
+            user.is_active = True
+            user.save()
+            user_profile, _ = UserProfile.objects.get_or_create(user=user)
+            user_profile.phone = request.data.get('phone', '')
+            user_profile.save()
+            # dict to setup language
+            update = {
+                u'pref-lang': language,
+            }
+            #setup language for user
+            update_user_preferences(user, update, username)
+            self.send_activation_email(request)
+        except ValidationError:
+            return Response(data={"error_message": "Wrong email format"}, status=status.HTTP_400_BAD_REQUEST)
+
+        return Response(data={'user_id': user.id, 'username': username}, status=status.HTTP_200_OK)
+
+    @staticmethod
+    def send_activation_email(request):
+        form = PasswordResetFormNoActive(request.data)
+        if form.is_valid():
+            form.save(use_https=request.is_secure(),
+                      from_email=configuration_helpers.get_value(
+                          'email_from_address', settings.DEFAULT_FROM_EMAIL),
+                      request=request,
+                      subject_template_name='sga_api/set_password_subject.txt',
+                      email_template_name='sga_api/set_password_email.html'
+                      )
+
+
+
+class BulkEnrollView(APIView, ApiKeyPermissionMixIn):
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = ApiKeyHeaderPermission,
+
+    def post(self, request):
+        """
+        Enrolling user on the course with the specified mod.
+        Create or update enroll user on the course. Can deactivate enroll
+        or activate enroll with use 'is_active' param.
+        """
+        data = request.data
+        users = data.get('users')
+        courses = data.get('courses')
+        action = data.get('action', 'enroll')
+        email_students = data.get('email_students', False)
+        auto_enroll = data.get('auto_enroll', False)
+
+        if action not in ['enroll', 'unenroll']:
+            error_msg = "action must be enroll or unenroll"
+        elif not isinstance(courses, (list, tuple)):
+            error_msg = "courses_id must be a list of courses_id "
+        elif not isinstance(users, (list, tuple)):
+            error_msg =  "users must be a list of identifiers(username or email)"
+        else:
+            error_msg = None
+
+        if error_msg:
+            return Response(
+                data={"error_message": error_msg},
+                status=status.HTTP_400_BAD_REQUEST
+            )
+
+        try:
+            result = self.__enroll_unenrol_users_for_courses(
```

Comment:
Two underscores starting the function's name is not a good choice, please use one.